### PR TITLE
CONSOLE-3616: Revert multicluster server flags

### PR DESCRIPTION
**Pull request by TheRealJon**
Decouple frontend and backend multicluster code by reverting multicluster server flag changes.

### CODE REVIEW

File: pkg/serverconfig/types.go [hunk 1]
```diff
@@ -269,8 +269,7 @@ type ManagedClusterOAuthConfig struct {
 // ManagedClusterConfig enables proxying to an ACM managed cluster
 // TODO remove multicluster
 type ManagedClusterConfig struct {
-	Name               string                        `json:"name" yaml:"name"`           // ManagedCluster name, provided through ACM
```
Comment (jhadvig) [inline pkg/serverconfig/types.go:272]: a lot more deleting awaits us 🥲

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
@TheRealJon we will need to rebase this PR

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
@TheRealJon looks like we will need to update the unite tests

**Comment by yapei** [comment]
covered following checkpoints in verification testing
prerequisite:  enable multi cluster feature and import some clusters as managed clusters
- check`window.SERVER_FLAGS.clusters` is removed
- check `window.SERVER_FLAGS.copiedCSVsDisabled` is updated to boolean: false
- check no cm/managed-clusters and other managed clusters configmaps generated
-  regression testing about `OperatorHub` and `Installed Operators` page
No issues found
/label qe-approved

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
@TheRealJon I got a feeling that this PR is enchanted by the rebase spell 🤣

**Review by jhadvig — APPROVED** [review]
/lgtm

**Pull request merged**